pkg/machinery/config/types/block/encryption.go

```diff
@@ -0,0 +1,289 @@
+	return &EncryptionSpec{
+		EncryptionProvider: block.EncryptionProviderLUKS2,
+		EncryptionKeys: []EncryptionKey{
+			{
+				KeySlot: 0,
+				KeyStatic: &EncryptionKeyStatic{
+					KeyData: "exampleKey",
+				},
+			},
+			{
+				KeySlot: 1,
+				KeyKMS: &EncryptionKeyKMS{
+					KMSEndpoint: "https://example-kms-endpoint.com",
+				},
+			},
+		},
+		EncryptionCipher:    "aes-xts-plain64",
+		EncryptionBlockSize: 4096,
+	}
+}
+
+// EncryptionKey represents configuration for disk encryption key.
+type EncryptionKey struct {
+	//   description: >
+	//     Key slot number for LUKS2 encryption.
+	KeySlot int `yaml:"slot"`
+	//   description: >
+	//     Key which value is stored in the configuration file.
+	KeyStatic *EncryptionKeyStatic `yaml:"static,omitempty"`
+	//   description: >
+	//     Deterministically generated key from the node UUID and PartitionLabel.
+	KeyNodeID *EncryptionKeyNodeID `yaml:"nodeID,omitempty"`
+	//   description: >
+	//     KMS managed encryption key.
+	KeyKMS *EncryptionKeyKMS `yaml:"kms,omitempty"`
+	//   description: >
+	//     Enable TPM based disk encryption.
+	KeyTPM *EncryptionKeyTPM `yaml:"tpm,omitempty"`
+}
+
+// EncryptionKeyStatic represents throw away key type.
+type EncryptionKeyStatic struct {
+	//   description: >
+	//     Defines the static passphrase value.
+	KeyData string `yaml:"passphrase,omitempty"`
+}
+
+// EncryptionKeyKMS represents a key that is generated and then sealed/unsealed by the KMS server.
+//
+//	examples:
+//	  - value: exampleKMSKey()
+type EncryptionKeyKMS struct {
+	//   description: >
+	//     KMS endpoint to Seal/Unseal the key.
+	KMSEndpoint string `yaml:"endpoint"`
+}
+
+// EncryptionKeyTPM represents a key that is generated and then sealed/unsealed by the TPM.
+type EncryptionKeyTPM struct {
+	//   description: >
+	//     Check that Secureboot is enabled in the EFI firmware.
+	//
+	//     If Secureboot is not enabled, the enrollment of the key will fail.
+	//     As the TPM key is anyways bound to the value of PCR 7,
+	//     changing Secureboot status or configuration
+	//     after the initial enrollment will make the key unusable.
+	TPMCheckSecurebootStatusOnEnroll *bool `yaml:"checkSecurebootStatusOnEnroll,omitempty"`
+}
+
+// EncryptionKeyNodeID represents deterministically generated key from the node UUID and PartitionLabel.
+type EncryptionKeyNodeID struct{}
+
+func exampleKMSKey() *EncryptionKeyKMS {
+	return &EncryptionKeyKMS{
+		KMSEndpoint: "https://192.168.88.21:4443",
+	}
+}
+
+// Validate implements config.Validator interface.
+//
+//nolint:gocyclo
+func (s EncryptionSpec) Validate() ([]string, error) {
+	if s.EncryptionProvider == block.EncryptionProviderNone && len(s.EncryptionKeys) == 0 {
+		return nil, nil
+	}
+
+	var errs error
+
+	switch s.EncryptionProvider {
+	case block.EncryptionProviderLUKS2:
+	case block.EncryptionProviderNone:
+		fallthrough
+	default:
+		errs = errors.Join(errs, fmt.Errorf("unsupported encryption provider: %s", s.EncryptionProvider))
```

Comment:
Won't this cause default to be always called? https://go.dev/wiki/Switch#fall-through

I might be wrong, this just feels a bit odd

Reply:
Go is different from C, it defaults not to fallthrough, so it should be equivalent to expect a single value to be ok

I put it this way, as the linter ensures I cover all type constant values, if we add more values in the future